File: src/main/java/by/kohanova/model/News.java
package by.kohanova.model;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Entity
@Table(name = "news")
public class News implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Integer id;

    @Column
    public String title;

    @ManyToOne
    @JoinColumn(name = "author")
    public User user;

    @Column
    public String content;

    @Column
    public Date date;
}

File: client/src/app/model/news.ts
import {User} from "./user";
interface INews {
    id?: number;
    title?: string;
    content?: string;
    author?: User;
    date?: Date;
}

export class News implements INews {
    id: number;
    title: string;
    content: string;
    author: User;
    date: Date;

    constructor(obj: INews = {}as INews) {
        let {
            title = '0',
            content = '0',
            author = null,

        }:INews = obj as INews;
        this.title = title;
        this.content = content;
        this.author = author;
    }
}